File: tests/fixtures/kat/cpp/taxonomy.cpp
// Comprehensive C++ feature taxonomy fixture.
// Exercises functions, classes, inheritance, templates, branches, throws,
// and call edges.

#include <stdexcept>
#include <string>

constexpr int CONSTANT = 42;

template <typename A, typename B>
struct Pair {
    A first;
    B second;
};

class Greeter {
public:
    virtual ~Greeter() = default;
    virtual std::string greet(const std::string &name) const = 0;
};

int add(int a, int b) {
    return a + b;
}

int branchy(int value, const std::string &mode) {
    int total = 0;
    if (value > 0) {
        total += value;
    } else if (value < 0) {
        total -= value;
    }
    for (int i = 0; i < value; i++) {
        if (i % 2 == 0) {
            total += i;
        }
    }
    while (total > 100) {
        total /= 2;
    }
    try {
        return std::stoi(mode) + total;
    } catch (const std::invalid_argument &) {
        throw std::runtime_error("bad mode");
    }
}

class Animal {
protected:
    std::string name_;
    std::string species_;

public:
    Animal(std::string name) : name_(std::move(name)), species_("unknown") {}
    virtual ~Animal() = default;

    std::string describe() const {
        return name_ + " the " + species_;
    }
};

class Dog : public Animal, public Greeter {
private:
    std::string breed_;

public:
    Dog(std::string name, std::string breed)
        : Animal(std::move(name)), breed_(std::move(breed)) {
        species_ = "dog";
    }

    std::string bark(bool loud) const {
        if (loud) {
            throw std::runtime_error("too loud");
        }
        return name_ + ": woof";
    }

    std::string greet(const std::string &name) const override {
        return name_ + " greets " + name;
    }
};

std::string use_animal(const Dog &d) {
    return d.bark(false);
}

File: tests/fixtures/kat/c_sharp/taxonomy.cs
// Comprehensive C# feature taxonomy fixture.

using System;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace Taxonomy
{
    public interface IGreeter
    {
        string Greet(string name);
    }

    public enum Status
    {
        Pending,
        Done
    }

    public static class Math
    {
        public const int Constant = 42;

        public static int Add(int a, int b)
        {
            return a + b;
        }

        public static int Branchy(int value, string mode)
        {
            int total = 0;
            if (value > 0)
            {
                total += value;
            }
            else if (value < 0)
            {
                total -= value;
            }
            for (int i = 0; i < value; i++)
            {
                if (i % 2 == 0)
                {
                    total += i;
                }
            }
            while (total > 100)
            {
                total /= 2;
            }
            try
            {
                return int.Parse(mode) + total;
            }
            catch (FormatException)
            {
                throw new InvalidOperationException("bad mode");
            }
        }

        public static async Task<string> FetchAsync(string url)
        {
            await Task.Yield();
            return url;
        }
    }

    public class Animal
    {
        public string Name { get; set; }
        public string Species { get; set; } = "unknown";

        public Animal(string name)
        {
            Name = name;
        }

        public virtual string Describe()
        {
            return $"{Name} the {Species}";
        }
    }

    public class Dog : Animal, IGreeter
    {
        public string Breed { get; set; }

        public Dog(string name, string breed = null) : base(name)
        {
            Species = "dog";
            Breed = breed;
        }

        public string Bark(bool loud = false)
        {
            if (loud)
            {
                throw new InvalidOperationException("too loud");
            }
            return $"{Name}: woof";
        }

        public string Greet(string name)
        {
            return $"{Name} greets {name}";
        }
    }
}
